crates/oxc_linter/examples/queries/next_no-head-import-in-document/match-_document-then-index.yml

```diff
@@ -0,0 +1,130 @@
+name: "next:no-head-import-in-document:match-_document-then-index"
+
+query: |
+  query {
+    File {
+      import {
+        from_path @filter(op: "=", value: ["$nextslashhead"])
+        span_: entire_span {
+          start @output
+          end @output
+        }
+      }
+
+      last_path_part {
+        name @filter(op: "has_prefix", value: ["$index"]) # TODO: this shouldn't be has_prefix, it really should be = to the file name, and ignore the extension
+        prev {
+          name @filter(op: "=", value: ["$underscore_document"])
+          prev {
+            name @filter(op: "=", value: ["$pages"])
+          }
+        }
+      }
+    }
+  }
+
+args:
+  nextslashhead: "next/head"
+  underscore_document: "_document"
+  pages: "pages"
+  index: "index"
+
+# TODO: replace ${document} with the file name
+reason: "`next/head` should not be imported in `pages${document}`. Use `<Head />` from `next/document` instead."
```

Comment:
it's a bug that we don't interpolate the string here still, not sure how important this is for first release